### PR TITLE
Expose proxies by default

### PR DESCRIPTION
**Pull request by nxtum**
Exposes all proxies by default when no id is specified via: Controller.1.proxies

### CODE REVIEW

**Review by sebaszm — COMMENTED** [review]
how about std::vector::insert  ?

**Review by sebaszm — DISMISSED** [review]

**Comment by pwielders** [comment]
@nxtum / @sebaszm there is probably a bit of a bigger story here. Lifetime :-) There was no addref in the past there is no addref now on the list that we passout, but Ithink it is a bit ticky (a larger flaw) The proxies are reference counted but as we pass them out here we do not increase the reference count which could be "scary"... Lets discuss tomorrow in the Sync up call..

**Comment by sebaszm** [comment]
> @nxtum / @sebaszm there is probably a bit of a bigger story here. Lifetime :-) There was no addref in the past there is no addref now on the list that we passout, but Ithink it is a bit ticky (a larger flaw) The proxies are reference counted but as we pass them out here we do not increase the reference count which could be "scary"... Lets discuss tomorrow in the Sync up call..

Ah, I didn't consider that.

File: Source/Thunder/PluginHost.cpp [hunk 1]
```diff
@@ -712,15 +712,13 @@ POP_WARNING()
                                 printf("Link: %s\n", index.Current().Remote.Value().c_str());
                                 printf("------------------------------------------------------------\n");
 
-                                RPC::Administrator::Proxies proxies;
-
-                                RPC::Administrator::Instance().Allocations(index.Current().ID.Value(), proxies);
-
-                                for (const ProxyStub::UnknownProxy* proxy : proxies) {
+                                RPC::Administrator::Instance().Allocations(index.Current().ID.Value(), [](const std::vector<ProxyStub::UnknownProxy*>& proxies) {
+                                    for (const auto& proxy: proxies) {
```
Comment (pwielders) [inline Source/Thunder/PluginHost.cpp:716]: Is this to test and see if I pay attention ;-) Please remove the printfs, and fill the proxies, cause that is a bit essential  ;-)

Comment (nxtum) [reply Source/Thunder/PluginHost.cpp:716]: Hi Pierre, the printfs are from existinng code:
https://github.com/rdkcentral/Thunder/blob/master/Source/Thunder/PluginHost.cpp#L721

Comment (nxtum) [reply Source/Thunder/PluginHost.cpp:716]: and I am not sure what you mean by filling the proxies:
the old code includes this:
RPC::Administrator::Proxies proxies; 

but i removed it as this part of the function only displays data, so I just directly use: 
[](const std::vector<ProxyStub::UnknownProxy*>& proxies)

I dont think there is a need to pass something like a new vector like
[&proxies]?
Or am i missing something?

Comment (pwielders) [reply Source/Thunder/PluginHost.cpp:716]: Ohhh sorry. You are right. My bad. This is the console print. I thought it was the JSONRPC call. Sorry!

**Review by pwielders — APPROVED** [review]

**Review by sebaszm — APPROVED** [review]

**Pull request merged**